package/MDAnalysis/core/topologyattrs.py

```diff
@@ -2041,6 +2051,412 @@ def total_charge(group, compound='group'):
+            where ``n`` is the number of compounds.
+
+
+        .. versionadded:: 2.4.0
+        """
+
+        compound = compound.lower()
+
+        atomgroup = group.atoms
+        charges = atomgroup.charges
+
+        if center == "mass":
+            masses = atomgroup.masses
+            ref = atomgroup.center_of_mass(
+                wrap=wrap, 
+                unwrap=unwrap, 
+                compound=compound
+            )
+        elif center == "charge":
+            ref = atomgroup.center_of_charge(
+                wrap=wrap,
+                unwrap=unwrap,
+                compound=compound
+            )
+        else:
+            choices = ["mass", "charge"]
+            raise ValueError(
+                "The dipole center, {}, is not supported. Choose one of: {}".format(
+                 center,choices))
+
+        if compound == 'group':
+            if wrap:
+                recenteredpos = (atomgroup.pack_into_box(inplace=False) - ref)
+            elif unwrap:
+                recenteredpos = (atomgroup.unwrap(inplace=False,
+                                                  compound=compound,
+                                                  reference=None,
+                                                 ) - ref)
+            else:
+                recenteredpos = (atomgroup.positions - ref)
+            dipole_vector = np.sum(recenteredpos * charges[:,np.newaxis],
+                                   axis=0)
+        else:
+            (atom_masks, 
+             compound_masks, 
+             n_compounds) = atomgroup._split_by_compound_indices(compound)
+
+            if unwrap:
+                coords = atomgroup.unwrap(
+                    compound=compound,
+                    reference=None,
+                    inplace=False
+                )
+            else:
+                coords = atomgroup.positions
+            chgs = atomgroup.charges
+
+            dipole_vector = np.empty((n_compounds, 3), dtype=np.float64)
+            for compound_mask, atom_mask in zip(compound_masks, atom_masks):
+                dipole_vector[compound_mask] = np.sum(
+                    (coords[atom_mask] - ref[compound_mask][:,None,:])
+                    * chgs[atom_mask][:,:,None], axis=1)
+
+        return dipole_vector
+
+    transplants[GroupBase].append(
+        ('dipole_vector', dipole_vector))
+
+    def dipole_moment(group, **kwargs):
+        r"""Dipole moment of the group or compounds in a group.
+
+        .. math::
+            \mu = \sqrt{ \sum_{i=1}^{D} \mathbf{\mu}^2 }
+
+        Where :math:`D` is the number of dimensions.
+
+        Computes the dipole moment of :class:`Atoms<Atom>` in the group.
+        Dipole per :class:`Residue`, :class:`Segment`, molecule, or
+        fragment can be obtained by setting the `compound` parameter
+        accordingly.
+
+        Note that when there is a net charge, the magnitude of the dipole 
+        moment is dependent on the ``center`` chosen. 
```

Comment:
Reference kwargs in single backtics
```suggestion
        moment is dependent on the `center` chosen. 
```